### ISSUE TITLE
executable bits not set

### ISSUE THREAD

**Issue by watilde** [watilde/dep#22]
// i will follow up the information

### Summary
> executable bits not set (what I originally found) https://github.com/watilde/dep/issues/7

### Steps to reproduce behavior
-

### Expected behavior
-

### Actual behavior
-

**Comment by gibfahn** [comment]
cc/ @sam-github 

I couldn't reproduce this (Ubuntu 16.04):

![image](https://user-images.githubusercontent.com/15943089/30208475-7cb7ab08-948b-11e7-9095-ed49a482f792.png)

<details><summary> Text version in case you want to copy/paste</summary>

```bash
❯ dep i gulp
Resolving dependencies
Installing dependencies
Installed 211 packages in 5.753s
❯ ls -l node_modules/.bin/gulp
lrwxrwxrwx 1 gib gib 51 Sep  8 11:45 node_modules/.bin/gulp -> /home/gib/tmp/deptest/node_modules/gulp/bin/gulp.js
```

</details>

**Comment by sam-github** [comment]
@gib Did you try to run gulp from .bin? The links have all permissions, but it doesn't matter, its the permissions of `.../bin/gulp.js` that matter when the link is followed.

**Comment by watilde** [comment]
Here is what I tried to reproduce, and it worked.

##### package.json
<details><summary>click here</summary>

```json
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "gulp": "gulp"
  },
  "keywords": [],
  "author": "",
  "license": "MIT"
}
```
</details>


##### Install
<details><summary>click here</summary>

```console
$ ls
package.json

$ dep install gulp
Resolving dependencies
Installing dependencies
Installed 212 packages in 7.598s

$ ls -la node_modules/ | grep gulp
drwxr-xr-x   11 watilde  staff   374 Sep  8 17:35 gulp
drwxr-xr-x    8 watilde  staff   272 Sep  8 17:35 gulp-util
drwxr-xr-x    7 watilde  staff   238 Sep  8 17:35 gulplog
drwxr-xr-x    6 watilde  staff   204 Sep  8 17:35 has-gulplog

$ ls -la node_modules/.bin/ | grep gulp
lrwxr-xr-x    1 watilde  staff    62 Sep  8 17:35 gulp -> /Users/watilde/Development/app/node_modules/gulp/bin/gulp.js
```
</details>

##### run
<details><summary>click here</summary>

```console
$ dep run gulp
[17:38:18] No gulpfile found
```

**Comment by sam-github** [comment]
I can reproduce on a clone of https://github.com/strongloop/strong-cluster-control

```
sn/strong-cluster-control (master u=) % nvm i 8
v8.4.0 is already installed.
npm install -gNow using node v8.4.0 (npm v5.3.0)
sn/strong-cluster-control (master u=) % npm install -g dep
...
sn/strong-cluster-control (master u=) % git clean -x -d
sn/strong-cluster-control (master u=) % dep i
Resolving dependencies
Installing dependencies
Installed 342 packages in 22.872s
sn/strong-cluster-control (master u=) % dep run pretest
(node:24042) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: /bin/sh: 1: eslint: Permission denied

(node:24042) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js proc
ess with a non-zero exit code.
sn/strong-cluster-control (master u=) % ls -l node_modules/.bin 
...
lrwxrwxrwx   1 sam sam    71 Sep  8 09:07 eslint -> /home/sam/w/sn/strong-cluster-control/node_modules/eslint/bin/eslint.js
sn/strong-cluster-control (master u=) % ls -l /home/sam/w/sn/strong-cluster-control/node_modules/eslint/bin/eslint.js
-rw-rw-r-- 1 sam sam 2762 Apr 29  2016 /home/sam/w/sn/strong-cluster-control/node_modules/eslint/bin/eslint.js
sn/strong-cluster-control (master u=) % uname -a
Linux samtu 4.8.0-59-generic #64-Ubuntu SMP Thu Jun 29 19:38:34 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
sn/strong-cluster-control (master u=) % ./node_modules/.bin/eslint
zsh: permission denied: ./node_modules/.bin/eslint
```

I  tried to check https://github.com/strongloop/strong-deploy with dep to see if it has the same problem, but dep seems to take many minutes to install.

**Comment by gibfahn** [comment]
I did try running it, and it was fine. Have you tried just `dep i eslint`?

```bash
❯ node --version
v8.4.0
❯ dep i eslint
Resolving dependencies
Installing dependencies
Installed 159 packages in 4.963s
❯ ./node_modules/.bin/eslint                                                                                                                                                                         ~/tmp/deptest
eslint [options] file.js [file.js] [dir]
[...]
```

**Comment by gibfahn** [comment]
Okay, so it fails for me on `strong-cluster-control`:

```bash
❯ g cl git@github.com:strongloop/strong-cluster-control.git                                                                                                                                                  ~/tmp
Cloning into 'strong-cluster-control'...
remote: Counting objects: 1042, done.
remote: Total 1042 (delta 0), reused 0 (delta 0), pack-reused 1042
Receiving objects: 100% (1042/1042), 200.01 KiB | 744.00 KiB/s, done.
Resolving deltas: 100% (603/603), done.
❯ ls                                                                                                                                                                                                         ~/tmp
bx  deptest  strong-cluster-control  swag
❯ cd strong-cluster-control                                                                                                                                                                                  ~/tmp
❯ dep i                                                                                                                                                                      ~/tmp/strong-cluster-control (master)
Resolving dependencies
Installing dependencies
Installed 340 packages in 9.036s
❯ ls                                                                                                                                                                         ~/tmp/strong-cluster-control (master)
api.md  CHANGES.md  CONTRIBUTING.md  docs.json  index.js  lib  LICENSE.md  Makefile  node_modules  package.json  README.md  test
❯ ./node_modules/.bin/eslint                                                                                                                                                 ~/tmp/strong-cluster-control (master)
zsh: permission denied: ./node_modules/.bin/eslint 
```

**Comment by watilde** [comment]
Hey I could reproduce and found the reason!


The original bin file at eslint@2.13.1 is not executable:
```console
$ wget http://registry.npmjs.org/eslint/-/eslint-2.13.1.tgz
$ tar -xf eslint-2.13.1.tgz
$ ls -l package/bin/
total 8
-rw-r--r--  1 watilde  staff  2762 Apr 29  2016 eslint.js
```

I will write a patch to chmod the permission then.

**Comment by watilde** [comment]
The patch to fix this issue was released as v0.14.6. Thanks!

**Comment by sam-github** [comment]
Thanks!
